orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java

```diff
@@ -105,172 +105,120 @@ public long getTimeout() {
   @Nonnull
   @Override
   public TaskResult execute(@Nonnull StageExecution stage) {
-    Map<String, String> rollbackTypes = new HashMap<>();
-    Map<String, Map> rollbackContexts = new HashMap<>();
-
     StageData stageData = stage.mapTo(StageData.class);
-    Map<String, Object> cluster;
-
     AtomicReference<Moniker> moniker = new AtomicReference<>(stageData.moniker);
-    if (moniker.get() == null && stageData.serverGroup != null) {
-      try {
-        Map<String, Object> serverGroup =
-            retrySupport.retry(
-                () ->
-                    fetchServerGroup(
-                        stageData.credentials, stageData.regions.get(0), stageData.serverGroup),
-                5,
-                1000,
-                false);
 
-        moniker.set(objectMapper.convertValue(serverGroup.get("moniker"), Moniker.class));
-      } catch (Exception e) {
-        logger.warn(
-            "Failed to fetch server group, retrying! (account: {}, region: {}, serverGroup: {})",
-            stageData.credentials,
-            stageData.regions.get(0),
-            stageData.serverGroup,
-            e);
-        return TaskResult.RUNNING;
-      }
+    boolean success =
+        populateMonikerWithServerGroupInfo(
+            moniker, stageData.credentials, stageData.regions.get(0), stageData.serverGroup);
+    if (!success) {
+      return TaskResult.RUNNING;
     }
 
-    try {
-      cluster =
-          retrySupport.retry(
-              () ->
-                  fetchCluster(
-                      moniker.get().getApp(),
-                      stageData.credentials,
-                      moniker.get().getCluster(),
-                      stageData.cloudProvider),
-              5,
-              1000,
-              false);
-    } catch (Exception e) {
-      logger.warn(
-          "Failed to fetch cluster, retrying! (application: {}, account: {}, cluster: {}, cloudProvider: {})",
-          moniker.get().getApp(),
-          stageData.credentials,
-          moniker.get().getCluster(),
-          stageData.cloudProvider,
-          e);
+    List<ServerGroup> serverGroups =
+        getServerGroups(moniker, stageData.credentials, stageData.cloudProvider);
+    if (serverGroups == null) {
       return TaskResult.RUNNING;
     }
 
-    List<ServerGroup> serverGroups =
-        objectMapper.convertValue(
-            cluster.get("serverGroups"), new TypeReference<List<ServerGroup>>() {});
-    serverGroups.sort(Comparator.comparing((ServerGroup o) -> o.createdTime).reversed());
+    return getResult(stageData, moniker.get().getCluster(), serverGroups);
+  }
+
+  private TaskResult getResult(
```

Comment:
Consider naming this after what it actually does, something like `determineRollbackCandidates`.

Reply:
Fixed in d074dd424